mlir/lib/Conversion/GPUToROCDL/LowerGpuOpsToROCDLOps.cpp

```diff
@@ -160,6 +160,27 @@ struct GPUSubgroupSizeOpToROCDL : ConvertOpToLLVMPattern<gpu::SubgroupSizeOp> {
   const amdgpu::Chipset chipset;
 };
 
+struct GPUBroadcastLaneOpToROCDL
+    : public ConvertOpToLLVMPattern<gpu::BroadcastLaneOp> {
+  using ConvertOpToLLVMPattern::ConvertOpToLLVMPattern;
+
+  LogicalResult
+  matchAndRewrite(gpu::BroadcastLaneOp op, OpAdaptor adaptor,
+                  ConversionPatternRewriter &rewriter) const override {
+    Value src = adaptor.getSrc();
+    if (adaptor.getBroadcastType() == gpu::BroadcastType::lane) {
+      rewriter.replaceOpWithNewOp<ROCDL::ReadlaneOp>(op, src.getType(), src,
+                                                     adaptor.getLane());
+    } else { // first_lane or any_lane
+      // any_lane is lowered to readfirstlane too, to force value into scalar
+      // register.
+      rewriter.replaceOpWithNewOp<ROCDL::ReadfirstlaneOp>(op, src.getType(),
```

Comment:
As the `any_lane` semantics got lost during lowering, I assume that `any_lane` is designed to help MLIR optimization at GPU dialect level and target-independent. 

Could you please describe what is the optimization passes you plan to do for `any_lane`, and what is the use case you target for?

Reply:
`any_lane` allows freely speculate across the control flow, see [mlir/test/Dialect/GPU/broadcast-speculatability.mlir](https://github.com/llvm/llvm-project/pull/152808/files#diff-b8af69e992ecd02399fa57abbdc056c4ec894ee23a9d095d3c714eca3725a7c8) test